pkg/logging/logger.go

```diff
@@ -47,35 +73,31 @@ const (
 	LogLevelError
 )
 
-func init() {
-	mutex.Lock()
-	defaultLogConsumer = NewFilteredLevelLogConsumer(LogLevelInfo, os.Stdout)
-	mutex.Unlock()
-}
-
 // SetLogger replaces the default logger with the given logger
-func SetLogger(logger OptimizelyLogConsumer) {
-	mutex.Lock()
-	defaultLogConsumer = logger
-	mutex.Unlock()
+func SetLogger(ctx context.Context, logger OptimizelyLogConsumer) {
+		setLogger(GetSdkKey(ctx), logger)
 }
 
 // SetLogLevel sets the log level to the given level
-func SetLogLevel(logLevel LogLevel) {
-	mutex.Lock()
-	defaultLogConsumer.SetLogLevel(logLevel)
-	mutex.Unlock()
+func SetLogLevel(ctx context.Context, logLevel LogLevel) {
+	if key := GetSdkKey(ctx); key != "" {
+		if logger := getLogger(key); logger != nil {
+			logger.SetLogLevel(logLevel)
+		}
+	}
 }
 
 // GetLogger returns a log producer with the given name
-func GetLogger(name string) OptimizelyLogProducer {
+func GetLogger(ctx context.Context, name string) OptimizelyLogProducer {
 	return NamedLogProducer{
+		ctx: ctx,
 		fields: map[string]interface{}{"name": name},
```

Comment:
The main crux of this PR is to make the SDK available to the OptimizelyLogConsumer. The `fields` member of the `NamedLogProducer` is meant to capture this type of arbitrary KV pairs to "enrich" the logs.

See https://github.com/optimizely/agent/blob/master/pkg/optimizely/logger.go#L54-L56 for usage of fields in an implementation of a OptimizelyLogConsumer.